spec/integration/bisect_spec.rb

```diff
@@ -33,14 +33,25 @@ def bisect(cli_args, expected_status=nil)
       end
     end
 
-    context "when the bisect commasaturingnd is long" do
+    context "when the bisect command saturates the pipe" do
       # On OSX and Linux a file descriptor limit meant that the bisect process got stuck at a certain limit.
       # This test demonstrates that we can run large bisects above this limit (found to be at time of commit).
       # See: https://github.com/rspec/rspec-core/pull/2669
       it 'does not hit pipe size limit and does not get stuck' do
         output = bisect(%W[spec/rspec/core/resources/blocking_pipe_bisect_spec.rb_], 1)
         expect(output).to include("No failures found.")
       end
+
+      it 'does not leave zombie processes', :unless => RSpec::Support::OS.windows? do
+        original_pids = pids()
+        bisect(%W[spec/rspec/core/resources/blocking_pipe_bisect_spec.rb_], 1)
+        sleep 0.1 while (extra_pids = pids() - original_pids).join.match?(/[RE]/i)
```

Comment:
Why the regex `[RE]`?

Reply:
`R` (running) and `E` (ended?) are valid process states before `Z` appears. Processes need to have cycled out of these state(s) before they will be seen as zombie.

Reply:
Thanks. Maybe declare this into a constant?
Something like:
```suggestion
        RUNNING_OR_ENDED_PROCESS = /[RE]/i
        sleep 0.1 while (extra_pids = pids() - original_pids).join.match?(RUNNING_OR_ENDED_PROCESS)
```

No problem if you want to keep it as it is.